app/src/main/java/src/Castling.java

```diff
@@ -0,0 +1,96 @@
+package src;
+
+import src.Pieces.Piece;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Castling implements MovePiece{
+
+    @Override
+    public List<Position> getPossibleMoves(Piece piece) {
+        List<Position> movePositions = new ArrayList<>();
+        int xPosQ = 0;
+        int xPosK = Chessboard.DIMENSION;
+
+        if(piece.getColorWhite()) {
+
+        }
+        if(!piece.isMoved()) {
+            //Castling King
+            int checkEmpty = 0;
+            if(!isHaveMoved(Chessboard.board[piece.getInitPosY()][xPosK])) {
+                for(int i = 1; i <= 2; i++) {
+                    if(!isSpaceEmpty(Chessboard.board[piece.getInitPosY()][xPosK - i])) {
+                        break;
+                    }
+                    checkEmpty++;
+                }
+                if(checkEmpty == 2)
+                    movePositions.add(new Position(xPosK - 1, piece.getInitPosY()));
+            }
+
+            //Castling Queen
+            if(!isHaveMoved(Chessboard.board[piece.getInitPosY()][xPosQ])) {
```

Comment:
Maybe refactor castling validations to another method and get boundaries as parameters.